packages/nimble-components/src/table/template.ts

```diff
@@ -22,7 +22,10 @@ export const template = html<Table>`
     <template role="table" ${children({ property: 'childItems', filter: elements() })}>
         <div class="table-container">
```

Comment:
I alluded to in another comment but we can rely on css property inheritance to separate out the bindings to the model from the template structure.

So we can do from one spot:
```html
<div class="table-container" style="
    --ni-private-table-header-row-grid-columns: ${x => x.rowGridColumns} auto;
    --ni-private-table-scroll-x: -${x => x.scrollX}px;
    --ni-private-header-scrollbar-spacer-width: ${x => x.virtualizer.headerContainerMarginRight}px;
    --ni-private-table-scroll-height: ${x => x.virtualizer.allRowsHeight}px;
    --ni-private-table-row-container-top: ${x => `${x.virtualizer.rowContainerYOffset}px; 
    --ni-private-table-row-grid-columns: ${x.rowGridColumns ?? ''}`
">
```

Helps reduce the noise in the template and makes a very clear list of the model state you can "bind" to via css properties for the styles.

One thought is do we need both `--ni-private-table-header-row-grid-columns` and `--ni-private-table-row-grid-columns`? Seems like we could just have a `--ni-private-table-row-grid-columns: ${x => x.rowGridColumns}` and account for the other tweaks on that in the css.